hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -650,15 +653,26 @@ public void deregisterAllListener(String cacheNameWithPrefix) {
     @Override
     public Map<String, LocalCacheStats> getStats() {
         Map<String, LocalCacheStats> stats = createHashMap(statistics.size());
+
         for (Map.Entry<String, CacheStatisticsImpl> entry : statistics.entrySet()) {
-            stats.put(entry.getKey(), new LocalCacheStatsImpl(entry.getValue()));
+            String cacheName = entry.getKey();
+            LocalCacheStatsImpl localCacheStats = new LocalCacheStatsImpl(entry.getValue());
+            CompletableFuture<CacheConfig> configFuture = configs.get(cacheName);
+            if (configFuture.isDone()) {
+                try {
+                    localCacheStats.setNativeMemoryUsed(configFuture.get().getInMemoryFormat() == NATIVE);
```

Comment:
Same as above, we either should send the in-memory format or rename to `persistentMemoryUsed` and set it to `true` when persistent memory is enabled.

Reply:
The problem is that at this point we only know that native memory is used, which is not necessarily persistent memory. So I took the approach of sending the in-memory format. So on MC side we will have to use both the root-level `TimedMemberState#persistendMemoryUsed` _and_ the `#inMemoryFormat` fields of data structures when we determine if the data structure uses optane memory or not.